packages/dvilela/skills/memeooorr_abci/behaviour_classes/twitter.py

```diff
@@ -320,90 +328,161 @@ def async_act(self) -> Generator:
         """Do the act, supporting asynchronous execution."""
 
         with self.context.benchmark_tool.measure(self.behaviour_id).local():
-            event = yield from self.get_event()
-
-            payload = EngageTwitterPayload(
-                sender=self.context.agent_address,
-                event=event,
-            )
+            event, new_mech_requests = yield from self.get_event()
+
+            if new_mech_requests:
+                mech_requests = json.dumps(new_mech_requests, sort_keys=True)
+                self.context.logger.info(f"Mech Requests JSON: {mech_requests}")
+
+            # add new prop to payload for tx_submitter and check for it in posttxround
+            if event == Event.MECH.value:
+                payload = EngageTwitterPayload(
+                    sender=self.context.agent_address,
+                    event=event,
+                    mech_request=json.dumps(new_mech_requests, sort_keys=True),
+                    tx_submitter=self.matching_round.auto_round_id(),
+                )
+            else:
+                payload = EngageTwitterPayload(
+                    sender=self.context.agent_address,
+                    event=event,
+                    mech_request=None,
+                    tx_submitter=self.matching_round.auto_round_id(),
+                )
```

Comment:
Check capitalization in comments. Why duplicate the full instance instead of just having a conditional mech_request object?